### PR TITLE
Feature/repositories and services

### PR DESCRIPTION
**Pull request by usingtechnology**
quite a huge shuffle of code, trying to get more organized and separate out concerns so we can build out the services.

### CODE REVIEW

**Review by Jsyro — COMMENTED** [review]
I drafted these yesterday, and your discord message addressed someone of them. we can chat about it today. Just didn't want to lose these notes.

File: services/traction/api/db/migrations/env.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    """
+    connectable = engine_from_config(
+        config.get_section(config.config_ini_section),
+        prefix="sqlalchemy.",
+        poolclass=pool.NullPool,
+    )
+
+    with connectable.connect() as connection:
+        context.configure(connection=connection, target_metadata=target_metadata)
```
[the remark this answers is not in the document]
Comment (usingtechnology) [reply services/traction/api/db/migrations/env.py:59]: sounds good, will add back in.

File: services/traction/api/db/repositories/access_keys.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    def _in_schema(self) -> Type[InAccessKeySchema]:
+        return InAccessKeySchema
+
+    @property
+    def _schema(self) -> Type[AccessKeySchema]:
+        return AccessKeySchema
+
+    @property
+    def _table(self) -> Type[AccessKey]:
+        return AccessKey
```
[the remark this answers is not in the document]
Comment (usingtechnology) [reply services/traction/api/db/repositories/access_keys.py:21]: will refactor using sql model to reduce duplication between tables and schemas. we discussed the purpose of repos at the meeting, but basically, yeah, isolate db transaction/session, reduce duplication.

File: services/traction/api/db/base_class.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import uuid
+from datetime import datetime
+
+from sqlalchemy import Column, func
+from sqlalchemy.dialects.postgresql import UUID, TIMESTAMP
+from sqlalchemy.ext.declarative import as_declarative, declared_attr
+
+
+@as_declarative()
+class Base:
```
[the remark this answers is not in the document]
Comment (usingtechnology) [reply services/traction/api/db/base_class.py:10]: yup, gonna move to use sql model.

**Comment by Jsyro** [comment]
Looks good.

**Pull request merged**